openaz-xacml/src/main/java/org/apache/openaz/xacml/std/jaxp/JaxpRequest.java

```diff
@@ -114,19 +123,13 @@ public static JaxpRequest load(File fileXmlRequest) throws ParserConfigurationEx
             logger.error("No Document returned parsing \"" + fileXmlRequest.getAbsolutePath() + "\"");
             return null;
         }
-
-        NodeList nodeListRoot = document.getChildNodes();
-        if (nodeListRoot == null || nodeListRoot.getLength() == 0) {
-            logger.warn("No child elements of the XML document");
-            return null;
+        
+        Node nodeRoot = document.getDocumentElement();
+        
+        if(context == null) {
+            init();
```

Comment:
Because every call to JAXBContext.newInstance will create a new instance of JAXBContext which is very expensive. According to [Unofficial JAXB Guide - Performance and thread-safety — Project Kenai](https://jaxb.java.net/guide/Performance_and_thread_safety.html), JAXBContext is thread safe and there should be only 1 instance for the whole application.

Also [Blaise Doughan](http://stackoverflow.com/users/383861/blaise-doughan)'s stackover flow [answer](http://stackoverflow.com/questions/7400422/jaxb-creating-context-and-marshallers-cost#answer-7400735) supports this.